test/parallel/test-stream2-writable.js

```diff
@@ -328,7 +328,7 @@ test('dont end while writing', function(t) {
     assert(!this.writing);
     wrote = true;
     this.writing = true;
-    setTimeout(function() {
+    setImmediate(function() {
```

Comment:
If you do `process.nextTick(process.nextTick(setImmediate(` you should virtually guarantee adding it to the *end* of the next processing of immediates, fwiw.